src/Ledger/GovernanceActions.lagda

```diff
@@ -67,7 +68,9 @@ data GovAction : Type where
 
 actionWellFormed : GovAction → Type
 actionWellFormed (ChangePParams x)  = ppdWellFormed x
-actionWellFormed (TreasuryWdrl x)   = ∀[ a ∈ dom x ] RwdAddr.net a ≡ NetworkId
+actionWellFormed (TreasuryWdrl x)   = 
+  ∀[ a ∈ dom x ] RwdAddr.net a ≡ NetworkId
+  × ¬ (∑ˢ[ v ← range x ] v ≡ 0)
```

Comment:
Wouldn't this be more clear as a "positive" property: `\forall v \in range x. \neg (v = 0)?

Reply:
That would be a slightly different statement, this one is stating that the sum of withdrawals is non-zero, but yours is saying that there can't be any zero coin withdrawals. Your condition makes more sense in my opinion, but the implementation looks at the sum and at this point we can't really change that, so the spec has to behave like that.

Reply:
Yeah, it's a good point. If you're checking that the withdrawals are sensible then forbidding zeros in the map makes more sense. For now we don't have a choice here, but we could make a ticket for the next era to change it.

Reply:
well, I guess I could write it as `∃[ v ∈ range x ] ¬ (v ≡ 0)`

Reply:
Ah, true, and then the proposed future change would be to replace `∃` by `∀`.